### PR TITLE
constify graphtile and others. add test for edge search. add filter function interface

### PR DESCRIPTION
**Pull request by kevinkreiser**

### CODE REVIEW

File: test/search.cc [hunk 1]
```diff
@@ -144,20 +144,50 @@ boost::property_tree::ptree make_tile() {
...
+    if(expected_name != tile->GetNames(tile->directededge(p.edges().front().id)->edgedataoffset())[0])
+      throw std::runtime_error("Didn't find expected road name");
+}
 
+void TestEdgeSearch() {
   auto conf = make_tile();
   valhalla::baldr::GraphReader reader(conf);
 
-  node_search(reader, {{.01, .2}}, {.01, .2});
+  //edge_search(reader, {{.095, .1}}, {.095, .1}, "3", .5);
```
Comment (kevinkreiser) [inline test/search.cc]: for now so the tests pass it doesnt do the edge search because since its unimplemented it fails
